exonum/src/runtime/rust/interfaces.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Important interservice communication interfaces.
+
+use crate::runtime::{dispatcher::Error as DispatcherError, MethodId};
+
+use super::{ExecutionError, Interface, TransactionContext};
+
+pub const INITIALIZE_METHOD_ID: MethodId = 0;
+
+/// This trait describes a service interface to pass initial configuration parameters into
+/// the started service instance.
+pub trait Initialize {
+    /// Initialize a service instance with the given parameters.
+    ///
+    /// The configuration parameters passed to the method are discarded immediately.
+    /// So the service instance should save them by itself if it is important for
+    /// the service business logic.
+    ///
+    /// This method is called after creating a new service instance by the [`start_service`]
+    /// invocation. In this case if an error during this action occurs, the dispatcher will
+    /// invoke [`stop_service`].
+    ///
+    ///
+    ///
+    /// ['start_service`]: ../../trait.Runtime.html#start_service
+    /// ['stop_service`]: ../../trait.Runtime.html#stop_service
+    fn initialize(&self, context: TransactionContext, params: &[u8]) -> Result<(), ExecutionError>;
```

Comment:
Why the method accepts `&[u8]`, given the fact that runtime probably wants to clone them and the Dispatcher won't use them after deserialization?

Reply:
Because the `call` method passes arguments as` & [u8] `and then they are passed unchanged to the interface.